### PR TITLE
engine: add 23.0.1 release notes

### PR DESCRIPTION
**Pull request by neersighted**
### Proposed changes

Proposed 23.0.1 release notes

### Related issues (optional)

- [docker/cli, 23.0.1 milestone](https://github.com/docker/cli/milestone/73?closed=1)
- [moby/moby, 23.0.1 milestone](https://github.com/moby/moby/milestone/113?closed=1)

### CODE REVIEW

File: engine/release-notes/23.0.md [hunk 1]
```diff
@@ -25,6 +25,31 @@ Changing the version format is a stepping-stone towards Go module compatibility,
...
+- [docker/cli, 23.0.1 milestone](https://github.com/docker/cli/milestone/73?closed=1)
+- [moby/moby, 23.0.1 milestone](https://github.com/moby/moby/milestone/113?closed=1)
+
+## Bug fixes and enhancements
+- Fix the daemon not starting if the kernel has AppArmor enabled, but `apparmor_parser` is not available. [moby/moby#44942](https://github.com/moby/moby/pull/44942)
+- Fix BuildKit-enabled builds with inline caching causing the daemon to crash. [moby/moby#44944](https://github.com/moby/moby/pull/44944)
+- Fix BuildKit improperly loading cached layers created by previous versions. [moby/moby#44959](https://github.com/moby/moby/pull/44959)
+- Fix upgrading with previously created `ipvlan` networks preventing the daemon from starting. [moby/moby#44937](https://github.com/moby/moby/pull/44937)
+- Fix the `overlay2` storage driver failing early in `metacopy` testing when initialized on an unsupported backing filesystem. [moby/moby#44922](https://github.com/moby/moby/pull/44922)
+- Fix exec exit events being misinterpreted as container exits under some runtimes, such as Kata Containers. [moby/moby#44892](https://github.com/moby/moby/pull/44892)
```
Comment (dvdksn) [inline engine/release-notes/23.0.md:43]:
```suggestion
- Fix `exec` exit events being misinterpreted as container exits under some runtimes, such as Kata Containers. [moby/moby#44892](https://github.com/moby/moby/pull/44892)
```

File: engine/release-notes/23.0.md [hunk 2]
```diff
@@ -25,6 +25,31 @@ Changing the version format is a stepping-stone towards Go module compatibility,
...
+- Improve the error message returned by the CLI when recieving a truncated JSON response when the API hangs up mid-request. [docker/cli#4004](https://github.com/docker/cli/pull/4004)
```
Comment (dvdksn) [inline engine/release-notes/23.0.md:44]:
```suggestion
- Improve the error message returned by the CLI when recieving a truncated JSON response caused by the API hanging up mid-request. [docker/cli#4004](https://github.com/docker/cli/pull/4004)
```

File: engine/release-notes/23.0.md [hunk 3]
```diff
@@ -25,6 +25,31 @@ Changing the version format is a stepping-stone towards Go module compatibility,
...
+- Fix an incorrect CLI exit code when attempting to execute a directory with a `runc` compiled using Go 1.20. [docker/cli#4004](https://github.com/docker/cli/pull/4004)
+- Fix mishandling the size argument to `--device-write-bps` as a path. [docker/cli#4004](https://github.com/docker/cli/pull/4004)
+
+## Packaging
```
Comment (dvdksn) [inline engine/release-notes/23.0.md:48]: add newline here
```suggestion
## Packaging

```

**Review by dvdksn — COMMENTED** [review]
Thanks for raising this PR ❤️  A couple suggestions

**Comment by neersighted** [comment]
Review comments addressed.

**Review by dvdksn — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM


I'll merge once the packages are up and the repos are tagged

**Comment by thaJeztah** [comment]
Release is up; https://github.com/moby/moby/releases/tag/v23.0.1

**Pull request merged**